thorlcr/graph/thgraphmaster.cpp

```diff
@@ -540,9 +540,26 @@ void CMasterActivity::getEdgeStats(IStatisticGatherer & stats, unsigned idx)
 void CMasterActivity::done()
 {
     CActivityBase::done();
+
+    StatsScopeId rootScope(SSTfile, 0U);
+    Owned<IStatisticGatherer> fileStats;
     ForEachItemIn(s, readFiles)
     {
         IDistributedFile &file = readFiles.item(s);
+        StringBuffer fileName;
+        file.getLogicalName(fileName);
+        rootScope.setFileId(fileName.str());
+        fileStats.setown(createStatisticsGatherer(SCTall, "", rootScope));
```

Comment:
There should probably be different helper functions to allow you to walk through CRuntimeStatistics